### ISSUE TITLE
Faster search and data processing

### ISSUE THREAD

**Issue by PEMapModder** [LegendOfMCPE/PocketFactions#7]
We need to search factions in a faster method when we search by:
- chunk
- player

We also need to improve the efficiency of calculating factino power.

**Comment by PEMapModder** [comment]
Fixing in branch `Db-Rewrite`

**Comment by iJoshuaHD** [comment]
about calculating powers

get the amount of time the player is online (i think StatsCore could do that) and the faction gets more powers by winning many kills and looses powers if the death of its members counts. 

about efficient searching through players and chunks, ill check your db rewrite for other suggestions.

i hope that helps a little.

**Comment by PEMapModder** [comment]
So your algorithm for power is?

Maybe this?

``` shell
SET FACTION_POWER=
  %TOTAL_ONLINE_SECONDS% * %AMPLIFIER_AS_IN_CONFIG% 
  + %PLAYER_KILLS% * %PLAYER_KILLS_AMPLIFIER %
  + %MOB_KILLS% * %AMPLIFIER%
  - %PLAYER_DEATHS% * %AMPLIFIER%
  - %BLAH%
```

Then,

``` PHP
if($chunkClaimFee <= $factionBalance and $factionPower >= $powerNeededForEachClaim * (count($claimedChunks) + 1)) return true:
else return false;
```

**Comment by iJoshuaHD** [comment]
yea :)

total time (secs) \* constant percentage + etc ...
= power of a faction :)

**Comment by PEMapModder** [comment]
And the one below?

**Comment by iJoshuaHD** [comment]
just make it or instead of and because of some reasons :P i guess you get it :)

``` PHP
if($chunkClaimFee <= $factionBalance or $factionPower >= $powerNeededForEachClaim * (count($claimedChunks) + 1)) return true:
else return false;
```

**Comment by PEMapModder** [comment]
Use `and`. Otherwise, people who get a lot of money can claim many chunks. Or, people who have always been AFK online (anti: idle kick?) will get a lot of power and become able to claim many chunks.

**Comment by iJoshuaHD** [comment]
well, as what i have understand, and operator needs both of the function while in or, either of the function is only needed.

**Comment by PEMapModder** [comment]
@ijoshuahd yes, so use and.

**Comment by PEMapModder** [comment]
Sorry I meant and not or

**Comment by PEMapModder** [comment]
Fixed in #8 obviously.
